### ISSUE TITLE
low mAP as the low batch_p?

### ISSUE THREAD

**Issue by guoxiaolu** [VisualComputingInstitute/triplet-reid#35]
I have trained the Market1501 dataset, and only change the batch_p to 12 as the OOM of my GPU. The final evaluation result is truly low as below:
mAP: 47.96% | top-1: 65.53%
I have also evaluate the pretrained model and the result is close to the given result (mAP: 66.88% | top-1: 83.40%).
I have noticed that someone using batch_p to 18 still get the reasonable result.  Is my result caused by the low batch_p?
`"batch_k": 4,
  "batch_p": 12,
  "checkpoint_frequency": 1000,
  "crop_augment": false,
  "decay_start_iteration": 15000,
  "detailed_logs": false,
  "embedding_dim": 128,
  "experiment_root": "./experiments/my_experiment",
  "flip_augment": false,
  "head_name": "fc1024",
  "image_root": "/home/tensor-2/triplet-reid/Market-1501-v15.09.15",
  "initial_checkpoint": null,
  "learning_rate": 0.0003,
  "loading_threads": 8,
  "loss": "batch_hard",
  "margin": "soft",
  "metric": "euclidean",
  "model_name": "resnet_v1_50",
  "net_input_height": 256,
  "net_input_width": 128,
  "pre_crop_height": 288,
  "pre_crop_width": 144,
  "resume": false,
  "train_iterations": 25000,
  "train_set": "data/market1501_train.csv"
`

The training step seems normal except the convergency is later than given plot.

**Comment by Pandoro** [comment]
This is possible, I don't actually recall us trying batch_p 12. I'd expect
it to be a little better than your result, but yes, for lower batch_p, the
result typically decreased. This is simply based on the fact that the hard
examples the network will see are less hard since you search for hard
examples in smaller batches. Additionally, you should read the README a
little more carefully. Your setup didn't use flip or crop augmentation,
thus you also didn't evaluate with flip/crop test-time augmentation. I'm
quite sure you will gain well over 10% from that. At least the test-time
augmentation gives about 5% and I expect that having no augmentation during
training will give you at least the same bump. Then you would end up
somewhere around 60%, which I say, is reasonable for batch_p 12.

If memory is the issue, my intuition would also be that batch_p 16 and
batch_k 3 might give you better results, but I wouldn't bet on it. Might be
worth a shot nevertheless.

**Comment by guoxiaolu** [comment]
Thank you, I'll check it

**Comment by lucasb-eyer** [comment]
The new result in your last comment is only enabling flip/crop augmentation, or also something else?

**Comment by guoxiaolu** [comment]
@lucasb-eyer , this is my args.json
`{
  "batch_k": 3,
  "batch_p": 16,
  "checkpoint_frequency": 1000,
  "crop_augment": true,
  "decay_start_iteration": 15000,
  "detailed_logs": false,
  "embedding_dim": 128,
  "experiment_root": "/home/tensor-2/triplet-reid/experiments/my_experiment",
  "flip_augment": true,
  "head_name": "fc1024",
  "image_root": "/home/tensor-2/triplet-reid/Market-1501-v15.09.15",
  "initial_checkpoint": "/home/tensor-2/triplet-reid/models/resnet_v1_50.ckpt",
  "learning_rate": 0.0003,
  "loading_threads": 8,
  "loss": "batch_hard",
  "margin": "soft",
  "metric": "euclidean",
  "model_name": "resnet_v1_50",
  "net_input_height": 256,
  "net_input_width": 128,
  "pre_crop_height": 288,
  "pre_crop_width": 144,
  "resume": false,
  "train_iterations": 25000,
  "train_set": "data/market1501_train.csv"
}
`
